core/api/gql/rollout.go

```
@@ -4,49 +4,133 @@ import (
+	return toGraphQLID(r.rollout.SelectorID)
 }
 
-type StaticRollout interface {
-	Rollout
-	AppVersion(ctx context.Context) AppVersion
+func (r Rollout) SelectorType() entity.SelectorType {
+	return r.rollout.SelectorType
 }
 
-type TimeRangeRollout interface {
-	Rollout
-	AppVersion(ctx context.Context) AppVersion
-	StartAt(ctx context.Context) *graphql.Time
-	EndAt(ctx context.Context) *graphql.Time
+func (r Rollout) ActivatorType() entity.ActivatorType {
+	return r.rollout.ActivatorType
 }
 
-type ExperimentRollout interface {
-	Rollout
-	AppVersions(ctx context.Context) []AppVersion
-	StartAt(ctx context.Context) *graphql.Time
-	EndAt(ctx context.Context) *graphql.Time
+func (r Rollout) IsEnabled() bool {
+	return r.rollout.IsEnabled
+}
+
+func (r Rollout) CreatedAt() graphql.Time {
+	return toGraphQLTime(r.rollout.CreatedAt)
+}
+
+func (r Rollout) UpdatedAt() *graphql.Time {
+	return toGraphQLTimePtr(r.rollout.UpdatedAt)
+}
+
+func (r Rollout) Activator(ctx context.Context) (Activator, error) {
+	switch r.rollout.ActivatorType {
+	case entity.ActivatorTypeTimeRange:
+		activator, err := r.deps.rolloutService.FindTimeRangeActivatorByID(ctx, r.rollout.ActivatorID)
+		if err != nil {
+			r.deps.logger.ErrorWithContext(ctx, err)
+			return nil, errs.ToResolverErr(err)
+		}
+
+		return newTimeRangeActivator(r.deps, activator), nil
+	case entity.ActivatorTypeMaxViewers:
+		activator, err := r.deps.rolloutService.FindMaxViewersActivatorByID(ctx, r.rollout.ActivatorID)
+		if err != nil {
+			r.deps.logger.ErrorWithContext(ctx, err)
+			return nil, errs.ToResolverErr(err)
+		}
+
+		return newMaxViewersActivator(r.deps, activator), nil
+	case entity.ActivatorTypePercentage:
+		activator, err := r.deps.rolloutService.FindPercentageActivatorByID(ctx, r.rollout.ActivatorID)
+		if err != nil {
+			r.deps.logger.ErrorWithContext(ctx, err)
+			return nil, errs.ToResolverErr(err)
+		}
+
+		return newPercentageActivator(r.deps, activator), nil
+	}
+
+	return nil, nil
```

I think we should throw some unsupported database error here.